modules/auxiliary/scanner/http/iis_shortname_scanner.rb

```diff
@@ -0,0 +1,280 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Auxiliary::Report
+  include Rex::Proto::Http
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name'        => 'Microsoft IIS shortname vulnerability scanner',
+        'Description' => %q{
+        'The vulnerability is caused by a tilde character "~" in a GET or OPTIONS request, which
+         could allow remote attackers to diclose 8.3 filenames (short names). This is a new
+         technique for an existing bug (CVE-2009-4444), which relied on the use of the ;(semicolon)
+         character. Soroush Dalili discovered the original bug, while the new ~ technique was
+         discovered by Soroush Dalili and Ali Abbasnejad. Older IIS installations are vulnerable
+         with GET, newer installations with OPTIONS'
+        },
+        'Author'         =>
+          [
+          'MinatoTW <shaks19jais[at]gmail.com>',
+          'egre55 <ianaustin[at]protonmail.com>'
+          ],
+        'License'     => MSF_LICENSE,
+        'References'     =>
+          [
+            [ 'CVE', '2009-4444' ],
+            [ 'URL', 'https://soroush.secproject.com/blog/tag/iis-tilde-vulnerability/' ]
+          ],
+        'Targets' => [[ 'Automatic', {} ]]
+      )
+    )
+
+  register_options([
+          Opt::RPORT(80),
+          OptString.new('PATH', [ true, "The base path to start scanning from", "/" ])
+
+  ])
+  @dirs = []
+  @files = []
+  @threads = []
+  @queue = Queue.new
+  @queue_ext = Queue.new
+  @alpha = 'abcdefghijklmnopqrstuvwxyz0123456789!#$%&\'()-@^_`{}'
+  @found = Array.new
+  @found2 = Array.new
+  @found3 = Array.new
+  @verb = ""
+  end
+
+  def check
+    begin
```

Comment:
It looks like this method is a duplication of the `is_vuln` method. Would it make more sense to call `is_vul` here, and return the relevant `CheckCode` depending on the Boolean response value from `is_vul` ?

Reply:
I agree, will do it